How-to-upgrade-review-version.md

```diff
@@ -0,0 +1,399 @@
+# How to Upgrade to TypeScript Without Anybody Noticing
+
+This guide will show you how to upgrade to Typescript without anybody
+noticing. Well, people *might* notice &mdash; what I really mean is
+that you won't have to change your build at all. You'll have the
+ability to get errors and completions in your editor and run `tsc`
+from the command line, but you won't have to integrate Typescript into
+your build.
+
+The way to do this is to not use Typescript. Stick with Javascript and
+use JSDoc to provide type information. This is less convenient than
+Typescript's syntax for types, but it means that your files stay
+boring old Javascript.
+
+Here's how. I'm going to use
+[typescript-eslint-parser](https://github.com/eslint/typescript-eslint-parser)
+as an example package so you can follow along if you want. Notably,
+this package *parses* Typescript, but is *written* in Javascript.
+
+## Add tsconfig
+
+I like to start with `tsc --init` and change the settings it gives you.
+
+I ended up with this:
+
+```json
+{
+  "compilerOptions": {
+    "target": "esnext",
+    "module": "commonjs",
+    "allowJs": true,
+    "checkJs": true,
+    "noEmit": true,
+    "resolveJsonModule": true
+    "esModuleInterop": true
+    "strict": false,
+  },
+  "exclude": [
+    "tests/fixtures",
+    "tests/integration"
+  ]
+}
+```
+
+For Javascript, your "compilerOptions" will pretty much always look
+like this. "resolveJsonModule" is optional, but you'll need it if your
+code ever `require`s a JSON file, so that Typescript will analyze it
+too. *Officially*, strict should be false, but as you will see later,
+ strict mode can be pretty great for Javascript code, not just
+ Typescript code.
+
+For typescript-eslint-parser, I added an exclude list. I wanted to
+check *all* the source files, because it's nice to have high
+aspirations. But it turns out the parser's tests take the form of JS
+files themselves, and I don't want to check those. Most of them are
+malformed in some way.
+
+You might want to use "include" if, say, you only want to check your
```

Comment:
With the exclude that you had in your tsconfig, the number of errors might be pretty overwhelming. Also, the whole process feels like an all-or-nothing thing - like the end goal of this tutorial is a huge commit with all the files in the repo being touched. I think you should explain how it's fine to work through at your own pace and that partial commits are perfectly fine since it doesn't affect any builds/tests.